File: apps/webflux/firstcome-firstserved/src/main/java/com/booster/firstcomefirstserved/common/response/ApiResponse.java
package com.booster.firstcomefirstserved.common.response;

public record ApiResponse<T>(
        boolean success,
        T data,
        String message
) {
    // 성공 응답 (데이터 있음)
    public static <T> ApiResponse<T> ok(T data) {
        return new ApiResponse<>(true, data, null);
    }

    // 성공 응답 (메시지만 있음 - 예: 주문 접수 등)
    public static <T> ApiResponse<T> accepted(String message) {
        return new ApiResponse<>(true, null, message);
    }

    // 실패 응답
    public static <T> ApiResponse<T> fail(String message) {
        return new ApiResponse<>(false, null, message);
    }
}

File: apps/kotlin/shopping-service/src/main/kotlin/com/booster/kotlin/shoppingservice/common/response/ApiResponse.kt
package com.booster.kotlin.shoppingservice.common.response


data class ApiResponse<T>(
    val success: Boolean,
    val data: T?,
    val error: ErrorResponse?,
) {
    companion object {
        fun <T> ok(data: T): ApiResponse<T> = ApiResponse(true, data, null)
        fun <T> fail(error: ErrorResponse): ApiResponse<T> = ApiResponse(false,null,error)
    }
}

data class ErrorResponse(
    val code: String,
    val message: String,
)